src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -8,25 +10,97 @@ namespace Orleans.Hosting
     public class EndpointOptions
     { 
         /// <summary>
-        /// The external IP address or host name used for clustering.
+        /// The IP address used for clustering.
         /// </summary>
-        public string HostNameOrIPAddress { get; set; }
```

Comment:
My concern with dropping this and only parsing when using the extension methods is that we drop support for declaratively configuring this (in appSettings.json, or XML or environment variables) so that it just works. Nevertheless of course the end user can explicitly parse this via the configuration subsystem and then pass it to the extension method, but there is a lot of extra juggling there...

Reply:
BTW, even though this wasn't "pure", this parsing was being done from the string in the EndpointOptions class instead of like the new proposed way, by design. I'm OK if you decide to change it consciously, but wanted to make sure the issue is understood before moving on with a more pure but less practical approach

---
In reply to: [167736599](https://github.com/dotnet/orleans/pull/4005#discussion_r167736599) [](ancestors = 167736599)